### DISCUSSION TITLE
In Firefox, how to avoid some security policies for local files?

### DISCUSSION THREAD

**Discussion by linuxmetel** [darkreader/darkreader#11687] [Q&A]
In Firefox, this add-on doesn't work for local files which loads local CSS files, you know. It's because of restriction on local file links and of the same-origin policy. The solution for Chrome is provided in [Dark Reader Help](https://darkreader.org/help/), but not for Firefox.
So I tried to solve it, and I know how to avoid the restriction on local file links. Here's how:

1. open `about:config`.
2. add three values: `capability.policy.policynames: localfilelinks`, `capability.policy.localfilelinks.checkloaduri.enabled: allAccess`, and `capability.policy.localfilelinks.sites: moz-extention://XXXX-XXXX-XXXX-XXXXXXXX/` (replace the URI of Dark Reader).

But I don't know how to avoid the same origin policy only for this addon (using `security.fileuri.strict_origin_policy` seems to be too risky for me). Is there any safe way to accomplish this?

And if there is a safe way to have Dark Reader work correctly for local files in Firefox, I think it should add into [Dark Reader Help](https://darkreader.org/help/).

**Comment by roke-julian-lockhart** [comment]
> In Firefox, this add-on doesn't work for local files which loads local CSS files

@linuxmetel, it works for me on local HTML files that link to local CSS files with `<link src="file:/path">`.

**Reply by linuxmetel** [reply]
Thank you for your comment! However, in most case, I open auto-generated HTML files, like documents for my Rust project, so...

**Reply by roke-julian-lockhart** [reply]
> https://github.com/darkreader/darkreader/discussions/11687#discussioncomment-11276234

@linuxmetel, but those shouldn't be fundamentally different to the browser. How are those auto-generated files linking to the CSS files, and where are the CSS files? To my knowledge, you can solely be using one of these things:

1. `<style>`
2. `<link rel="https://">`
3. `<link rel="file://">`

Surely we are discussing the last option? If so, I can't imagine you're doing anything special. This should work for you.

**Reply by linuxmetel** [reply]
I mean at least, I have to write a script or run `sed` in order to replace all of those in the auto-generated files (by Rustdoc or suchlike), and that each generation requires rerunning it. It seems easier to just run `python3 -m http.server` and access `localhost:PORT`.
